### PR TITLE
[WIP] Make Repeater name configurable

### PR DESCRIPTION
**Pull request by SmittieC**
## Technical Summary
See the [ticket](https://dimagi-dev.atlassian.net/jira/software/c/projects/SC/boards/196?modal=detail&selectedIssue=SC-2361) for the reason for this PR.

This is the third and last PR in a series of 3 that aims to add a name field to the Repeaters.
Quick summary: Now that the name attribute exists on the Repeater model, we can update it. This PR adds code that exposes a `name` field when the user creates a new Repeater or update an existing one. This field allows the user to either
1. Give the repeater a custom name
2. if left blank, the repeater name will default to its connection setting's name

See the screenshot below of what this new form looks like:
## When creating a repeater:
Screenshot from 2022-11-28 14-23-59

###
We can now see the new name (and it's not defaulting to "Connection 1")
Screenshot from 2022-11-28 14-25-19

## When updating a repeater:
Screenshot from 2022-11-28 14-25-46

This PR can only be merged once the [first PR](https://github.com/dimagi/commcare-hq/pull/32359) is deployed

## Safety Assurance
I tested adding, updating and removing repeaters locally. **Please note** that I still need to add unit tests

### Safety story
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by sravfeyn — APPROVED** [review]
👍

File: corehq/ex-submodules/dimagi/utils/couch/migration.py [hunk 1]
```diff
@@ -183,6 +183,8 @@ def _migration_do_sync(self):
         return sql_object
 
     def save(self, *args, **kwargs):
+        if not self.name:
+            self.name = self.connection_settings.name
```
Comment (sravfeyn) [inline corehq/ex-submodules/dimagi/utils/couch/migration.py:187]:
```suggestion
        self.name = self.name or self.connection_settings.name
```

File: corehq/motech/repeaters/models.py [hunk 2]
```diff
@@ -300,12 +302,6 @@ class Meta:
 
     _has_config = False
 
-    @property
-    def repeater_name(self):
-        # This is a temporary name change. We can't have the 'name' property and the name attribute at the same
-        # time. This method/property will be removed in a subsequent PR.
```
Comment (sravfeyn) [inline corehq/motech/repeaters/models.py:306]: 💯

File: docker/min_settings.py [hunk 3]
```diff
@@ -87,19 +87,19 @@
 
 BASE_ADDRESS = '{}:8000'.format(os.environ.get('HQ_PORT_8000_TCP_ADDR', 'localhost'))
 
-######## Email setup ########
+######## Email setup ######## noqa: E266
```
Comment (sravfeyn) [inline docker/min_settings.py:90]: What are these for?

**Review by sravfeyn — APPROVED** [review]

File: docker/min_settings.py (hunk 3 above)
Comment (SmittieC) [reply docker/min_settings.py:90]: There was a linting issue where it wasn't happy about all the #'s. This tells the lint checker to ignore this comment

**Pull request closed without merging**